### PR TITLE
Test for erc20

### PR DESCRIPTION
**Pull request by Violent-Idiot**

### CODE REVIEW

File: contracts/Aave.sol [hunk 1]
```diff
@@ -78,21 +82,37 @@ contract Aave {
         token.withdraw(amt);
     }
 
-    function deposit(address tokenaddr) external payable {
-        // address tokenaddr = weth;
-        bool isEth = (tokenaddr == eth) ? true : false;
-        tokenaddr = (isEth) ? weth : tokenaddr;
+    function ERCdeposit(address tokenaddr, uint256 amt) external payable {
+        console.log(address(this));
+        IAave aave = IAave(provider.getLendingPool());
```
Comment (pradyuman-verma) [inline contracts/Aave.sol:87]: No need to create this thing again and again!
Just create one global.

Comment (Violent-Idiot) [reply contracts/Aave.sol:87]: I will try to clean up redundant code after erc20 testing finishes.

File: contracts/Aave.sol [hunk 2]
```diff
@@ -78,21 +82,37 @@ contract Aave {
...
+        IERC20 token = IERC20(tokenaddr);
+        require(token.balanceOf(msg.sender) >= amt, "Insufficent funds");
+        console.log(token.balanceOf(msg.sender) / (10**18));
+        console.log(msg.sender);
+        token.approve(address(this), amt);
+        // token.allowance(msg.sender, address(this));
+        token.transferFrom(msg.sender, address(this), amt);
+        token.approve(address(aave), amt);
+        aave.deposit(address(tokenaddr), amt, address(this), 0);
```
Comment (pradyuman-verma) [inline contracts/Aave.sol:96]: You can't approve your own address for some amount!
You have to approve this in the script/test

File: test/sample-test.js [hunk 3]
```diff
@@ -1,36 +1,69 @@
...
       await hre.network.provider.request({
         method: "hardhat_impersonateAccount",
         params: ["0x15BB5127Bd75025AaCe941bC7296fCEEF46980b1"],
       });
-      // console.log(addr);
+      const provider = new ethers.providers.JsonRpcProvider(process.env.INFURA);
+      const daiAddress = "0x4F96Fe3b7A6Cf9725f59d353F723c1bDb64CA6Aa";
       const owner = await ethers.getSigner(
         "0x15BB5127Bd75025AaCe941bC7296fCEEF46980b1"
       );
```
Comment (pradyuman-verma) [inline test/sample-test.js:27]: const [owner] = await ethers.getSigner(
        "0x15BB5127Bd75025AaCe941bC7296fCEEF46980b1"
      );

File: test/sample-test.js [hunk 4]
```diff
@@ -1,36 +1,69 @@
...
+      let bal = await daiContract.balanceOf(
+        "0x15BB5127Bd75025AaCe941bC7296fCEEF46980b1"
+      );
+      console.log(bal, contractAddr.address);
+      daiContract
+        .transfer(contractAddr.address, ethers.utils.parseUnits("1", 0))
+        .then((res) => console.log(res))
+        .catch((err) => console.log(err));
+      console.log(owner.address, ethers.utils.parseUnits("1", 0));
+
```
Comment (pradyuman-verma) [inline test/sample-test.js:41]: You have to approve here!

Comment (Violent-Idiot) [reply test/sample-test.js:41]: Instead of transfer I have write approve right?

**Pull request merged**